### PR TITLE
[Mono] Intrinsify Narrow for for Vector{64, 128} on Arm64

### PR DESCRIPTION
**Pull request by fanyang-mono**
Contributes to https://github.com/dotnet/runtime/issues/64072

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -672,6 +672,28 @@ type_to_extract_op (MonoTypeEnum type)
 	}
 }
 
+static MonoClass *
+create_class_instance (MonoCompile *cfg, MonoMethodSignature *fsig, const char* name_space, const char *name, gboolean same_type, MonoType *param_type)
```
Comment (lambdageek) [inline src/mono/mono/mini/simd-intrinsics.c:676]: Maybe it's better to break this up into two functions:
1. `MonoType *vector_type_get_element_type (MonoType *vector_inst_type)`: given a `Vector<C>` it returns `C`.  (basically the code that pulls out `type_argv[0]) here).
2. `MonoClass * create_vector_instance (const char* name_space, const char *name, MonoType *param_type)`.

Then at the call sites:
```c
   MonoType *param_type = vector_type_get_element_type (fsig->params[0]);
   MonoClass *vector_inst = create_class_instance ("S.R.I", "Vector128`1", param_type);
```

(also: the `MonoCompile *cfg` arg is unused.)


---

Another thought:  should we add `vector128_class` and `vector64_class` to `mono_defaults` ? It seems like we look them up pretty often now

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:676]: I searched string "Vector128`1" in mono codebase. I don't see they were looked up at any other place.

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:676]: I would like to have this PR merged soon. Will submit a follow-up PR to split the function and remove the unused arg.

**Review by lambdageek — APPROVED** [review]

**Comment by fanyang-mono** [comment]
The failures on CI are irrelevant to this PR.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:676]: Resolved via https://github.com/dotnet/runtime/pull/67125

**Pull request merged**